proto/celestia/blob/v1/tx.proto

```diff
@@ -18,19 +18,14 @@ message MsgPayForBlobs {
   // signer is the bech32 encoded signer address. See
   // https://en.bitcoin.it/wiki/Bech32.
   string signer = 1;
-  // namespaces is a list of namespaces that the blobs are associated with. A
-  // namespace is a byte slice of length 29 where the first byte is the
-  // namespaceVersion and the subsequent 28 bytes are the namespaceId.
-  repeated bytes namespaces = 2;
+  // 2 was previously used for namespaces.
+  reserved 2;
   // blob_sizes is a list of blob sizes (one per blob). Each size is in bytes.
   repeated uint32 blob_sizes = 3;
   // share_commitments is a list of share commitments (one per blob).
   repeated bytes share_commitments = 4;
-  // share_versions are the versions of the share format that the blobs
-  // associated with this message should use when included in a block. The
-  // share_versions specified must match the share_versions used to generate the
-  // share_commitment in this message.
-  repeated uint32 share_versions = 8;
+  // 8 was previously used for share_versions;
+  reserved 8;
 }
```

Comment:
blocking

since this is breaking, do we need to create a v2 of the proto msg?

I think https://github.com/celestiaorg/celestia-app/issues/2932#issuecomment-1856474966 makes sense, and we might have to add an intermediate type to convert between v1 and v2 so that we don't have to have an extra copy of the logic to execute the msg.